### PR TITLE
EffectController

### PR DESCRIPTION
**Pull request by floschu**
Adds [EffectController](https://github.com/floschu/control/blob/feature/effect-type/control-core/src/main/kotlin/at/florianschuster/control/EffectController.kt).

Depends on https://github.com/floschu/control/pull/23

### CODE REVIEW

File: buildSrc/src/main/kotlin/Libs.kt [hunk 1]
```diff
@@ -181,6 +178,11 @@ object Libs {
     const val espresso_core: String = "androidx.test.espresso:espresso-core:" +
             Versions.espresso_core
 
+    /**
+     * https://github.com/Kotlin/atomicfu
```
Comment (AlexGrafl) [inline buildSrc/src/main/kotlin/Libs.kt:182]: link broken?

Comment (AlexGrafl) [reply buildSrc/src/main/kotlin/Libs.kt:182]: https://github.com/Kotlin/kotlinx.atomicfu

File: examples/android-github/src/main/kotlin/at/florianschuster/control/androidgithub/search/GithubViewModel.kt [hunk 2]
```diff
@@ -69,6 +75,10 @@ internal class GithubViewModel(
                         emit(Mutation.SetLoadingNextPage(true))
                         emitAll(
                             flow { emit(api.search(state.query, state.page + 1)) }
+                                .catch { error ->
+                                    Log.w("GithubViewModel.Query", error)
+                                    emitEffect(Effect.SearchError)
```
Comment (AlexGrafl) [inline examples/android-github/src/main/kotlin/at/florianschuster/control/androidgithub/search/GithubViewModel.kt:80]: maybe change one example to a try catch, as your api.search is a suspending function

File: buildSrc/src/main/kotlin/Libs.kt (hunk 1 above)
Comment (floschu) [reply buildSrc/src/main/kotlin/Libs.kt:182]: That is automatically generated by `buildSrcVersions` 🤷

File: examples/android-github/src/main/kotlin/at/florianschuster/control/androidgithub/search/GithubViewModel.kt (hunk 2 above)
Comment (floschu) [reply examples/android-github/src/main/kotlin/at/florianschuster/control/androidgithub/search/GithubViewModel.kt:80]: Can do 👍 
I think that I will actually create a whole different example for an `EffectController`.

**Pull request closed without merging**